pkg/cli/flags.go

```diff
@@ -946,29 +952,105 @@ func init() {
 	}
 }
 
-type tenantIDWrapper struct {
-	tenID *roachpb.TenantID
-}
-
-func (w *tenantIDWrapper) String() string {
-	return w.tenID.String()
-}
-func (w *tenantIDWrapper) Set(s string) error {
+func setTenantID(s string, cfgTenantID *roachpb.TenantID) error {
 	tenID, err := strconv.ParseUint(s, 10, 64)
 	if err != nil {
 		return errors.Wrap(err, "invalid tenant ID")
 	}
 	if tenID == 0 {
 		return errors.New("invalid tenant ID")
 	}
-	*w.tenID = roachpb.MustMakeTenantID(tenID)
+	*cfgTenantID = roachpb.MustMakeTenantID(tenID)
 	return nil
 }
 
+type tenantIDWrapper struct {
+	tenID *roachpb.TenantID
+}
+
+func (w *tenantIDWrapper) String() string {
+	return w.tenID.String()
+}
+func (w *tenantIDWrapper) Set(s string) error {
+	return setTenantID(s, w.tenID)
+}
+
 func (w *tenantIDWrapper) Type() string {
 	return "number"
 }
 
+//	fileFlag := mtStartSQLCmd.Flags().Lookup(cliflags.TenantIDFromFile.Name)
+//	if fileFlag == nil {
+//		return nil, errors.New("tenant id from file callback executed without tenant id file")
+//	}
+//	fileName := fileFlag.Value.String()
+
+// tenantIDFromFile will look for the given file and read the full first
+// line of the file that should contain the `<TenantID>`.
+func tenantIDFromFile(
+	fileName string, watcherWaitCount *atomic.Uint32, watcherEventCount *atomic.Uint32,
+) (*roachpb.TenantID, error) {
+
+	// Start watching the file for changes as the typical case is that the file
+	// will not have yet the tenant id at startup.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
```

Comment:
nit: constructing the watcher should annotate the error and all of the errors returned by the watcher should be annotated. From experience, its common for CI environments and dev machines to run out of inotify handles. The kernel errors are vague and difficult to root cause.